examples/v1/pipelineruns/alpha/pipelinerun-with-matrix-context-variables.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: echomatrixlength
+spec:
+  params:
+    - name: matrixlength
+      type: string
+  steps:
+    - name: echo
+      image: alpine
+      script: echo $(params.matrixlength)
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: echomatrixresultslength
+spec:
+  params:
+    - name: matrixresultslength
+      type: string
+  steps:
+    - name: echo
+      image: alpine
+      script: echo $(params.matrixresultslength)
```

Comment:
instead of echoing, we can confirm the expected length - similar to https://github.com/tektoncd/pipeline/blob/main/examples/v1/pipelineruns/alpha/pipelinerun-large-results.yaml#L75